packages/dashboard-frontend/src/store/DevfileRegistries/getEditor.ts

```diff
@@ -10,44 +10,69 @@
  *   Red Hat, Inc. - initial API and implementation
  */
 
+import { dump } from 'js-yaml';
 import { ThunkDispatch } from 'redux-thunk';
 
+import devfileApi from '@/services/devfileApi';
 import { actionCreators, KnownAction } from '@/store/DevfileRegistries/index';
 import { AppState } from '@/store/index';
 
 export async function getEditor(
   editorIdOrPath: string,
   dispatch: ThunkDispatch<AppState, unknown, KnownAction>,
   getState: () => AppState,
-  pluginRegistryUrl?: string,
 ): Promise<{ content?: string; editorYamlUrl: string; error?: string }> {
   let editorYamlUrl: string;
 
+  const state = getState();
+
   if (/^(https?:\/\/)/.test(editorIdOrPath)) {
     editorYamlUrl = editorIdOrPath;
-  } else {
-    if (!pluginRegistryUrl) {
-      throw new Error('Plugin registry URL is required.');
+    let devfileObj = state.devfileRegistries.devfiles[editorYamlUrl];
+    if (devfileObj) {
+      const content = devfileObj.content;
+      const error = devfileObj.error;
+      return Object.assign({ content, editorYamlUrl, error });
     }
-    editorYamlUrl = `${pluginRegistryUrl}/plugins/${editorIdOrPath}/devfile.yaml`;
-  }
-
-  const state = getState();
-  let devfileObj = state.devfileRegistries.devfiles[editorYamlUrl];
-  if (devfileObj) {
-    const content = devfileObj.content;
-    const error = devfileObj.error;
-    return Object.assign({ content, editorYamlUrl, error });
-  }
-  await dispatch(actionCreators.requestDevfile(editorYamlUrl));
+    await dispatch(actionCreators.requestDevfile(editorYamlUrl));
 
-  const nexState = getState();
-  devfileObj = nexState.devfileRegistries.devfiles[editorYamlUrl];
-  if (devfileObj) {
-    const content = devfileObj.content;
-    const error = devfileObj.error;
-    return Object.assign({ content, editorYamlUrl, error });
+    const nexState = getState();
+    devfileObj = nexState.devfileRegistries.devfiles[editorYamlUrl];
+    if (devfileObj) {
+      const content = devfileObj.content;
+      const error = devfileObj.error;
+      return Object.assign({ content, editorYamlUrl, error });
+    }
+    throw new Error(`Failed to fetch editor yaml by URL: ${editorYamlUrl}.`);
+  } else {
+    const editors = state.dwPlugins.cmEditors || [];
+    let editor: devfileApi.Devfile | undefined;
+    for (const e of editors) {
+      if (!e.metadata.name)
+        throw new Error(
+          `Missing metadata.name attribute in the editor yaml file: ${editorIdOrPath}.`,
+        );
+      if (!e.metadata.attributes.publisher)
+        throw new Error(
+          `Missing metadata.attributes.publisher attribute in the editor yaml file: ${editorIdOrPath}.`,
+        );
+      if (!e.metadata.attributes.version)
+        throw new Error(
+          `Missing metadata.attributes.version attribute in the editor yaml file: ${editorIdOrPath}.`,
+        );
```

Comment:
I think it would be better not to throw exceptions here, but rather do all the necessary validations and verifications right where you get these editors from the backend.

Reply:
fixed